dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -380,29 +376,25 @@ protected void notify(URL url, NotifyListener listener, List<URL> urls) {
             logger.warn("Ignore empty notify urls for subscribe url " + url);
             return;
         }
+        if (urls == null) {

```

Comment:
Order wise I think we should place this `if(urls == null)` before `if ((urls == null || urls.isEmpty())` , it will have more harmony in login layouting. Then you can replace the code `if ((urls == null || urls.isEmpty()) && !Constants.ANY_VALUE.equals(url.getServiceInterface()))`   with `if (urls.isEmpty() && !Constants.ANY_VALUE.equals(url.getServiceInterface()))`

Do you feel it will make logic bit clearer?